docs/2023/puzzles/day05.md

```diff
@@ -6,6 +6,151 @@ import Solver from "../../../../../website/src/components/Solver.js"
 
 https://adventofcode.com/2023/day/5
 
+## Solution summary
+
+First and foremost, the data must be parsed from the file into the following classes:
+- ```final case class Resource(start: Long, `type`: DestinationEnum)``` the source resource in the first task
+- ```final case class ResourceWithRange(start: Long, end: Long, `type`: DestinationEnum)``` the source resource in the second task, in the first one you can use this same model but set `end = 0`
+- I used the following enumeration for storing the type of resource:
+```scala
+object DestinationEnum extends Enumeration {
+  type DestinationEnum = Value
+  val Seed, Soil, Fertilizer, Water, Light, Temperature, Humidity, Location = Value
+}
+```
+- ```final case class ResourceMap(from: DestinationEnum, to: DestinationEnum, properties: Seq[Property])``` and ```final case class Property(startTo: Long, startFrom: Long, range: Long)``` for storing data about converting from one resource type to another
```

Comment:
Since property directly corresponds to the Advent of Code input, I'd name it morke like it is in the problem.

`startTo` is called "destination range start"
`startFrom` is called "source range start"
`range` is called "range length"

The `DestinationEnum`, `from` and `to` are called the "source category" and "destination category"

Do with that what you will.